installscripts/jazz-terraform-unix-noinstances/ecs.tf

```diff
@@ -0,0 +1,389 @@
+
+  default_action {
+    target_group_arn = "${aws_alb_target_group.alb_target_group_codeq.arn}"
+    type             = "forward"
+  }
+}
+
+data "aws_ecs_task_definition" "ecs_task_definition" {
+  count = "${var.dockerizedJenkins}"
+  task_definition = "${aws_ecs_task_definition.ecs_task_definition.family}"
+}
+
+data "aws_ecs_task_definition" "ecs_task_definition_gitlab" {
+  count = "${var.scmgitlab}"
+  task_definition = "${aws_ecs_task_definition.ecs_task_definition_gitlab.family}"
+}
+
+data "aws_ecs_task_definition" "ecs_task_definition_codeq" {
+  count = "${var.codeq * var.scmgitlab * var.dockerizedJenkins}"
+  task_definition = "${aws_ecs_task_definition.ecs_task_definition_codeq.family}"
+}
+
+resource "aws_ecs_service" "ecs_service" {
+  count = "${var.dockerizedJenkins}"
+  name            = "${var.envPrefix}_ecs_service"
+  task_definition = "${aws_ecs_task_definition.ecs_task_definition.family}:${max("${aws_ecs_task_definition.ecs_task_definition.revision}", "${data.aws_ecs_task_definition.ecs_task_definition.revision}")}"
+  desired_count   = 1
+  launch_type     = "FARGATE"
+  health_check_grace_period_seconds  = 3000
+  cluster =       "${aws_ecs_cluster.ecs_cluster.id}"
+
+  network_configuration {
+    security_groups    = ["${lookup(var.jenkinsservermap, "jenkins_security_group")}"]
+    subnets            = ["${lookup(var.jenkinsservermap, "jenkins_subnet")}", "${lookup(var.jenkinsservermap, "jenkins_subnet2")}"]
+    assign_public_ip = true
+  }
+
+  load_balancer {
+    target_group_arn = "${aws_alb_target_group.alb_target_group.arn}"
+    container_name   = "${var.envPrefix}_ecs_container"
+    container_port   = "8080"
+  }
+  provisioner "local-exec" {
+      command = "sleep 1m"
+  }
+  depends_on = ["aws_alb_target_group.alb_target_group", "aws_lb.alb_ecs"]
+}
+
+resource "aws_ecs_service" "ecs_service_gitlab" {
+  count = "${var.scmgitlab}"
+  name            = "${var.envPrefix}_ecs_service_gitlab"
+  task_definition = "${aws_ecs_task_definition.ecs_task_definition_gitlab.family}:${max("${aws_ecs_task_definition.ecs_task_definition_gitlab.revision}", "${data.aws_ecs_task_definition.ecs_task_definition_gitlab.revision}")}"
+  desired_count   = 1
+  launch_type     = "FARGATE"
+  health_check_grace_period_seconds  = 3000
+  cluster =       "${aws_ecs_cluster.ecs_cluster_gitlab.id}"
+
+  network_configuration {
+    security_groups    = ["${lookup(var.jenkinsservermap, "jenkins_security_group")}"]
+    subnets            = ["${lookup(var.jenkinsservermap, "jenkins_subnet")}", "${lookup(var.jenkinsservermap, "jenkins_subnet2")}"]
+    assign_public_ip = true
+  }
+
+  load_balancer {
+    target_group_arn = "${aws_alb_target_group.alb_target_group_gitlab.arn}"
+    container_name   = "${var.envPrefix}_ecs_container_gitlab"
+    container_port   = "80"
+  }
+  provisioner "local-exec" {
+      command = "sleep 4m"
+  }
+  depends_on = ["aws_alb_target_group.alb_target_group_gitlab", "aws_lb.alb_ecs_gitlab"]
```

Comment:
Sometimes getting this error, seems the load balancer is not ready.

* aws_ecs_service.ecs_service_gitlab: 1 error(s) occurred:

* aws_ecs_service.ecs_service_gitlab: InvalidParameterException: The target group with targetGroupArn arn:aws:elasticloadbalancing:us-east-XXXXXX:targetgroup/testecs2715-ecs-gitlab-tg/XXXXXXXXXXX does not have an associated load balancer.

Reply:
Good, now we know that's a bug with the Terraform AWS provider.

Other people have encountered this, it can be worked around without sleeps, using a single `depends_on` statement, as described here:

https://github.com/hashicorp/terraform/issues/12634#issuecomment-394921275

Reply:
you mean, add a single depends_on for the existing resource  - "aws_alb_target_group.alb_target_....."  or non-existent resource ["aws_lb_listener_rule.main"]

Reply:
In the linked bug there are several different approaches, but since the error is literally saying that the `target_group` doesn't have a load balancer ready yet, suggest you first try adding the ALB resource as a dependency inside each ALB Target Group resource to work around this bug.

We can remove the dependencies when this bug is fixed in the AWS provider, so please put a comment above each extra `depends_on` you add stating why we added it, and when it can be removed.

Reply:
Done